sonic-xcvrd/src/xcvrd_utilities/y_cable_helper.py

```diff
@@ -0,0 +1,510 @@
+            "Could not retreive fieldvalue pairs for {}, inside config_db".format(logical_port_name))
+        return
+
+    else:
+        # Convert list of tuples to a dictionary
+        mux_table_dict = dict(fvs)
+        if "mux_cable" in mux_table_dict:
+            if y_cable_presence[0] is True:
+                # delete this entry in the y cable table found and update the delete event
+                namespaces = multi_asic.get_front_end_namespaces()
+                for namespace in namespaces:
+                    asic_id = multi_asic.get_asic_index_from_namespace(
+                        namespace)
+                    state_db[asic_id] = daemon_base.db_connect(
+                        "STATE_DB", namespace)
+                    y_cable_tbl[asic_id] = swsscommon.Table(
+                        state_db[asic_id], swsscommon.STATE_HW_MUX_CABLE_TABLE_NAME)
+                # fill the newly found entry
+                delete_port_from_y_cable_table(
+                    logical_port_name, y_cable_tbl[asic_index])
+                delete_change_event[:] = [True]
+
+
+def init_ports_status_for_y_cable(platform_sfp, platform_chassis, y_cable_presence, stop_event=threading.Event()):
+    global y_cable_platform_sfputil
+    global y_cable_platform_chassis
+    # Connect to CONFIG_DB and create port status table inside state_db
+    config_db, state_db, port_tbl, y_cable_tbl = {}, {}, {}, {}
+    port_table_keys = {}
+
+    y_cable_platform_sfputil = platform_sfp
+    y_cable_platform_chassis = platform_chassis
+
+    # Get the namespaces in the platform
+    namespaces = multi_asic.get_front_end_namespaces()
+    for namespace in namespaces:
+        asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+        config_db[asic_id] = daemon_base.db_connect("CONFIG_DB", namespace)
+        port_tbl[asic_id] = swsscommon.Table(config_db[asic_id], "PORT")
+        port_table_keys[asic_id] = port_tbl[asic_id].getKeys()
+
+    # Init PORT_STATUS table if ports are on Y cable
+    logical_port_list = y_cable_platform_sfputil.logical
+    for logical_port_name in logical_port_list:
+        if stop_event.is_set():
+            break
+
+        # Get the asic to which this port belongs
+        asic_index = y_cable_platform_sfputil.get_asic_id_for_logical_port(
+            logical_port_name)
+        if asic_index is None:
+            helper_logger.log_warning(
+                "Got invalid asic index for {}, ignored".format(logical_port_name))
+            continue
+
+        if logical_port_name in port_table_keys[asic_index]:
+            check_identifier_presence_and_update_mux_table_entry(
+                state_db, port_tbl, y_cable_tbl, asic_index, logical_port_name, y_cable_presence)
+        else:
+            ''' This port does not exist in Port table of config but is present inside
+                logical_ports after loading the port_mappings from port_config_file
+                This should not happen
+            '''
```

Comment:
Suggest using comments (`#`) here instead of docstring. Docstrings are intended to document how to _use_ the code, whereas comments are indented to document how the code _works_.

Reply:
fixed